pkg/reconciler/buildrun/buildrun.go

```diff
@@ -210,6 +220,7 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 				serviceAccount := &corev1.ServiceAccount{}
 				serviceAccount.Name = resources.GetGeneratedServiceAccountName(buildRun)
 				serviceAccount.Namespace = buildRun.Namespace
+				buildRun.Status.ServiceAccountName = &serviceAccount.Name
```

Comment:
This one needs to go away. It will set the value we previously set during TaskRun creation. Is a no-op.